api/src/main/java/edu/cornell/mannlib/vitro/webapp/web/templatemodels/individual/GroupedPropertyList.java

```diff
@@ -136,10 +140,36 @@ is to make the get methods (getProperty and getPropertyAndRemoveFromList) method
         // Build the template data model from the groupList
         groups = new ArrayList<PropertyGroupTemplateModel>(
                 propertyGroupList.size());
+        // submit parallel
+        ExecutorService executor = Executors.newCachedThreadPool();
         for (PropertyGroup propertyGroup : propertyGroupList) {
-            groups.add(new PropertyGroupTemplateModel(vreq, propertyGroup,
-                    subject, editing, populatedDataPropertyList,
-                    populatedObjectPropertyList));
+            groups.add(new PropertyGroupTemplateModel(vreq, propertyGroup, subject, editing,
+                    populatedDataPropertyList, populatedObjectPropertyList, executor));
+        }
+        
+        // wait all
+        for (PropertyGroupTemplateModel pgtm : groups) {
+            List<Future<?>> futures = pgtm.getFutures();
```

Comment:
Would be better if this was the more precise List<Future<PropertyTemplateModel>>, but requires the Callable models to be the Callable<PropertyTemplateModel>.

Reply:
maybe we can make 

ObjectPropertyTemplateModelCallable implements Callable<PropertyTemplateModel>  
DataPropertyTemplateModelCallable implements Callable<PropertyTemplateModel>

then   

List<Future<PropertyTemplateModel>> futures = pgtm.getFutures();

Reply:
Ah - yes. That's better! (It's also what I typed into the review, but the editor must have taken it to be html and mangled it).

PropertyTemplateModel as the generic class throughout, rather than an indeterminate Future type.